static/app/stories/view/storyMdxComponent.tsx

```diff
@@ -1,14 +1,16 @@
+import {Text} from 'sentry/components/core/text';
+
 import {StoryHeading} from './storyHeading';
 
 type HeadingProps = {
   children: React.ReactNode;
 };
 
 export const storyMdxComponents = {
-  h1: (props: HeadingProps) => <StoryHeading as="h1" size="2xl" {...props} />,
-  h2: (props: HeadingProps) => <StoryHeading as="h2" size="2xl" {...props} />,
-  h3: (props: HeadingProps) => <StoryHeading as="h3" size="xl" {...props} />,
-  h4: (props: HeadingProps) => <StoryHeading as="h4" size="lg" {...props} />,
-  h5: (props: HeadingProps) => <StoryHeading as="h5" size="md" {...props} />,
-  h6: (props: HeadingProps) => <StoryHeading as="h6" size="sm" {...props} />,
+  h1: () => null,
```

Comment:
Could we instead just update the titles here to use the correct N+1 levels? I'm afraid that someone is going to inevitably use an H1 and wonder why nothing is rendered, or use a H6 and wonder why a paragraph is rendered. It will probably end up being a not so pleasant surprise or a pita to debug.